### PR TITLE
Deliverable 1 (2/2)

### PR DESCRIPTION
**Pull request by andreschabb**
1) Using create-react-app, what do we need to set up for testing?
Nothing, It has already installed all needed.

2) What components are worth to test in your development?
All components should be tested

3) Can you apply TDD once you already created components?
Yes, following the principles.

### CODE REVIEW

File: coverage.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+> npm run test -- --coverage --watchAll
+
+
+> react-certification-2020@0.1.0 test /Users/andres.chavez/Documents/Wizeline/react-certification-2021
+> react-scripts test "--coverage" "--watchAll"
+
+--------------------------|----------|----------|----------|----------|-------------------|
+File                      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
+--------------------------|----------|----------|----------|----------|-------------------|
+All files                 |    93.16 |    93.75 |    90.91 |    93.16 |                   |
```
Comment (wdonet) [inline coverage.txt:12]: Awesome! 👍🏽  Thanks for adding this

File: src/components/Header/Header.test.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    expect(userIcon[0]).toBeTruthy();
+  });
+
+  it('should opens login modal', () => {
+    const onOpenModal = jest.fn();
+
+    render(<Header onOpenModal={onOpenModal} />);
+
+    fireEvent.click(screen.getByTestId('header-dropdown'));
+    fireEvent.click(screen.getByTestId('header-dropdown-login'));
```
Comment (wdonet) [inline src/components/Header/Header.test.jsx:70]: This is a nice way to simulate user actions over unit tests 👌🏽

File: src/components/Layout/Layout.test.jsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  });
+  it('should renders', () => {
+    render(
+      <Layout>
+        <div id="content">Content</div>
+      </Layout>
+    );
+
+    const container = document.querySelector('#content');
+    expect(container).toBeTruthy();
```
Comment (wdonet) [inline src/components/Layout/Layout.test.jsx:33]: why not using a `data-test-id` in order to use a `getByTestId()` instead?
actually, why not using any other HTML element that can be reachable by role using react-testing-library?

File: src/components/Layout/Layout.test.jsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  });
+
+  it('should renders with the sidebar closed', () => {
+    render(
+      <Layout>
+        <div id="content">Content</div>
+      </Layout>
+    );
+
+    const sidebarOpened = document.body.getElementsByClassName(
```
Comment (wdonet) [inline src/components/Layout/Layout.test.jsx:45]: why not use `layout-sidebar` from  `data-testid` attribute instead?

File: src/components/Layout/Layout.test.jsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+  it('should opens the sidebar when clicking bar icon', () => {
+    render(
+      <Layout>
+        <div id="content">Content</div>
+      </Layout>
+    );
+
+    fireEvent.click(screen.getByTestId('header-bars-icon'));
+    const sidebarOpened = document.body.getElementsByClassName(
```
Comment (wdonet) [inline src/components/Layout/Layout.test.jsx:59]: same as above, this could be reachable via data test id attribute

Comment (wdonet) [reply src/components/Layout/Layout.test.jsx:59]: if you want to be sure it has exactly that list of class names, you can iterate over the `classList` prop from the resultant object

File: src/components/Login/Login.test.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    fireEvent.change(usernameInput, { target: { value: 'admin' } });
+    fireEvent.change(passwordInput, { target: { value: 'admin' } });
+    expect(usernameInput.value).toBe('admin');
+    expect(passwordInput.value).toBe('admin');
+
+    fireEvent.click(getByTestId('login-button'));
+    expect(usernameInput.value).toBe('');
+    expect(passwordInput.value).toBe('');
+  });
```
Comment (wdonet) [inline src/components/Login/Login.test.jsx:40]: Suggestion:
- another good expectation would be to check that the login fn was called once

File: src/components/Private/Private.test.jsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    useAuth.mockImplementation(() => ({ authenticated: true }));
+    render(
+      <MemoryRouter initialEntries={['/Private']}>
+        <Private>
+          <div id="private">Private</div>
+        </Private>
+      </MemoryRouter>
+    );
+
+    const container = document.querySelector('#private');
```
Comment (wdonet) [inline src/components/Private/Private.test.jsx:23]: Why not use `getByText('Private')` in this case?

**Review by wdonet — APPROVED** [review]
💯  Great job 👌🏽  on the test cases. It seems you explored well the react testing library to mock responses and functions to be called on certain contexts.

I just added a few suggestions but I think you're on a good track for unit testing your components

File: src/components/Login/Login.test.jsx (hunk 6 above)
Comment (andreschabb) [reply src/components/Login/Login.test.jsx:40]: Sure, I will add it. Thanks!

**Comment by andreschabb** [comment]
Thanks for the feedback, I will address these comments

**Pull request merged**